packages/contracts/deploy/3_upgrade_repo/30_upgrade_repo.ts

```diff
@@ -0,0 +1,134 @@
+import {PLUGIN_REPO_ENS_NAME} from '../../plugin-settings';
+import {PluginRepo__factory} from '../../typechain';
+import {
+  getProductionNetworkName,
+  getAragonDeploymentsInfo,
+} from '../../utils/helpers';
+import {
+  getLatestNetworkDeployment,
+  getNetworkNameByAlias,
+} from '@aragon/osx-commons-configs';
+import {PLUGIN_REPO_PERMISSIONS} from '@aragon/osx-commons-sdk';
+import {DeployFunction} from 'hardhat-deploy/types';
+import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import path from 'path';
+
+type SemVer = [number, number, number];
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const {network} = hre;
+  const [deployer] = await hre.ethers.getSigners();
+  const productionNetworkName: string = getProductionNetworkName(hre);
+
+  const pluginRepo = PluginRepo__factory.connect(
+    getAragonDeploymentsInfo(network.name)[network.name]['address'],
+    deployer
+  );
+
+  console.log(
+    `Upgrading plugin repo '${PLUGIN_REPO_ENS_NAME}.plugin.dao.eth' (${pluginRepo.address})...`
+  );
+
+  const newPluginRepoImplementation = PluginRepo__factory.connect(
+    getLatestNetworkDeployment(getNetworkNameByAlias(productionNetworkName)!)!
+      .PluginRepoBase.address,
+    deployer
+  );
+
+  // TODO Use the `getProtocolVersion` function from osx-commons-sdk
+  let current: SemVer;
+  try {
+    current = await pluginRepo.protocolVersion();
+  } catch {
+    current = [1, 0, 0];
+  }
+
+  const latest: SemVer = await newPluginRepoImplementation.protocolVersion();
+
+  console.log(
+    `Upgrading from current protocol version v${current[0]}.${current[1]}.${current[2]} to the new version v${latest[0]}.${latest[1]}.${latest[2]}.`
+  );
+
+  // Prepare optional initialization data
+
+  // TODO Add `initializeFrom` function to `PluginRepo`.
+  const initializeFromCalldata: any[] = [];
+  /*
+  const initData: any[] = [];
```

Comment:
rm this commented code if not needed

Reply:
I turned this into a more detailed note and explanation